src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -475,43 +1249,217 @@ private void slicePoints()
+                        {
+                            logger.error("Relation {} has a loop! Parent tree: {}",
+                                    subRelation.getIdentifier(), parents);
+                            slicedMembers = null;
+                        }
+
+                        if (slicedMembers != null)
+                        {
+                            this.statistics.recordSlicedRelation();
+                            slicedMembers.forEach(
+                                    slicedRelation -> members.add(new TemporaryRelationMember(
+                                            memberIdentifier, member.getRole(), memberType)));
+                        }
+                        else
+                        {
+                            members.add(temporaryMember);
+                        }
+                        break;
+                    default:
+                        // Raw Atlas should not have any other types than the ones here
+                        throw new CoreException("Unsupported {} Member for Relation {}", memberType,
+                                relation.getIdentifier());
+                }
+            }
+            else
+            {
+                // Found member to remove, remove it to shorten our list
+                removedMembers.remove(temporaryMember);
+            }
+        }
+
+        // Add in any new members as a result of multipolygon fixes
+        final List<TemporaryRelationMember> addedMembers = Optional.ofNullable(
+                this.slicedRelationChanges.getAddedRelationMembers().get(relation.getIdentifier()))
+                .orElse(new ArrayList<>());
+        addedMembers.forEach(newMember -> members.add(newMember));
+
+        // Group members by country
+        final Map<String, List<TemporaryRelationMember>> countryEntityMap = groupRelationMembersByCountry(
+                members);
+
+        final List<TemporaryRelationMember> membersWithoutCountry;
+        if (countryEntityMap.containsKey(ISOCountryTag.COUNTRY_MISSING))
+        {
+            membersWithoutCountry = countryEntityMap.remove(ISOCountryTag.COUNTRY_MISSING);
+        }
+        else
+        {
+            membersWithoutCountry = Collections.emptyList();
+        }
+
+        final int countryCount = countryEntityMap.size();
+
+        if (countryCount == 0)
+        {
+            // Assign a missing country
+            final Map<String, String> tags = new HashMap<>();
+            tags.put(ISOCountryTag.KEY, ISOCountryTag.COUNTRY_MISSING);
+            this.slicedRelationChanges.updateRelationTags(relation.getIdentifier(), tags);
+        }
+        else if (countryCount == 1)
+        {
+            // Assign the single country
+            final Map<String, String> tags = new HashMap<>();
+            tags.put(ISOCountryTag.KEY, countryEntityMap.keySet().iterator().next());
+            this.slicedRelationChanges.updateRelationTags(relation.getIdentifier(), tags);
```

Comment:
If it's possible there's a missing country tag, is it possible that the countryCount could really be 2 here? As in the relation crosses a boundary but one of the country tags is missing so the countryCount is 1.

Reply:
We remove all missing country code tags on line 1397, otherwise it would definitely be possible.